internal/commands/user/list/command.go

```diff
@@ -13,8 +20,142 @@ var Command = cli.CommandDefinition{
 	Command:     &command{},
 }
 
-type command struct{}
+type command struct {
+	inputs      inputs
+	realmClient realm.Client
+	users       []realm.User
+}
+
+func (cmd *command) Flags(fs *pflag.FlagSet) {
+	cmd.inputs.Flags(fs)
+
+	fs.VarP(&cmd.inputs.UserState, flagState, flagStateShort, flagStateUsage)
+	fs.BoolVarP(&cmd.inputs.Pending, flagPending, flagPendingShort, false, flagPendingUsage)
+	fs.StringSliceVarP(&cmd.inputs.ProviderTypes, flagProvider, flagProviderShort, []string{}, flagProviderUsage)
+	fs.StringSliceVarP(&cmd.inputs.Users, flagUser, flagUserShort, []string{}, flagUserUsage)
+}
+
+func (cmd *command) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+func (cmd *command) Setup(profile *cli.Profile, ui terminal.UI) error {
+	cmd.realmClient = realm.NewAuthClient(profile.RealmBaseURL(), profile.Session())
+	return nil
+}
 
 func (cmd *command) Handler(profile *cli.Profile, ui terminal.UI) error {
+	app, appErr := cli.ResolveApp(ui, cmd.realmClient, cmd.inputs.Filter())
+	if appErr != nil {
+		return appErr
+	}
+	users, usersErr := cmd.realmClient.FindUsers(
+		app.GroupID,
+		app.ID,
+		realm.UserFilter{
+			State:     cmd.inputs.UserState,
+			Pending:   cmd.inputs.Pending,
+			Providers: cmd.inputs.ProviderTypes,
+			IDs:       cmd.inputs.Users,
+		},
+	)
+	if usersErr != nil {
+		return usersErr
+	}
+
+	cmd.users = users
+
 	return nil
 }
+
+const (
+	headerID                     = "ID"
+	headerName                   = "Name"
+	headerEmail                  = "Email"
+	headerEnabled                = "Enabled"
+	headerType                   = "Type"
+	headerLastAuthenticationDate = "Last Authentication"
+
+	userDataEmail = "email"
+	userDataName  = "name"
+)
+
+func (cmd *command) Feedback(profile *cli.Profile, ui terminal.UI) error {
+	if len(cmd.users) == 0 {
+		return ui.Print(terminal.NewTextLog("No available users to show"))
+	}
+
+	var usersByProviderType = make(map[string][]realm.User)
+	for _, user := range cmd.users {
+		for _, identity := range user.Identities {
+			usersByProviderType[identity.ProviderType] = append(usersByProviderType[identity.ProviderType], user)
+		}
+	}
+
+	var logs []terminal.Log
+	for providerType, users := range usersByProviderType {
+		sortUsersByLastAuthentication(users)
+
+		logs = append(logs, terminal.NewTableLog(
+			fmt.Sprintf("Provider type: %s", providerType),
+			userTableHeaders(providerType),
+			userTableRows(providerType, users)...,
+		))
+	}
+	return ui.Print(logs...)
+}
+
+func sortUsersByLastAuthentication(users []realm.User) {
+	sort.Slice(
+		users,
+		func(i, j int) bool {
+			return users[i].LastAuthenticationDate > users[j].LastAuthenticationDate
+		},
+	)
+}
+
+func userTableHeaders(providerType string) []string {
```

Comment:
can we add a few quick unit tests for `userTableHeaders` and `userTableRow` (I'm thinking we're ok without unit tests for `userTableRows`, but your call as to how to slice it.)